src/agent/src/mount.rs

```diff
@@ -573,6 +581,92 @@ fn mount_storage(logger: &Logger, storage: &Storage) -> Result<()> {
     )
 }
 
+#[instrument]
+pub fn set_ownership(logger: &Logger, storage: &Storage) -> Result<()> {
+    let logger = logger.new(o!("subsystem" => "change_ownership"));
+
+    // If fsGroup is not set, skip performing ownership change
+    if storage.fs_group == 0 {
+        return Ok(());
+    }
+
+    let mut read_only = false;
+    let opts_vec: Vec<String> = storage.options.to_vec();
+    if opts_vec.contains(&String::from("ro")) {
+        read_only = true;
+    }
+
+    let mount_path = Path::new(&storage.mount_point);
+
+    let mut skip_ownership_change = false;
+    if storage.fs_group_change_policy == FSGroupChangePolicy::OnRootMismatch {
+        if mount_path.metadata()?.gid() == storage.fs_group {
+            let mut mask = RW_MASK;
+            if read_only {
+                mask = RO_MASK;
+            }
+            mask |= EXEC_MASK;
+
+            // With fsGroup change policy to OnRootMismatch, if the current
+            // gid of the mount path root directory matches the desired gid
+            // and the current permission of mount path root directory is correct,
+            // then ownership change will be skipped.
+            let current_mode = mount_path.metadata()?.permissions().mode();
+            if (mask & current_mode == mask) && (current_mode & MODE_SETGID != 0) {
+                info!(logger, "skipping ownership change for volume";
+                "mount-destination" => mount_path.to_str(),
+                "fs-group" => storage.fs_group.to_string(),
+            );
+                skip_ownership_change = true
+            }
+        }
+    }
+    if !skip_ownership_change {
+        info!(logger, "performing recursive ownership change";
+            "mount-destination" => mount_path.to_str(),
+            "fs-group" => storage.fs_group.to_string(),
+        );
+        recursive_ownership_change(
+            mount_path,
+            None,
+            Some(Gid::from_raw(storage.fs_group)),
+            read_only,
+        )?;
+    }
+
+    Ok(())
+}
+
+#[instrument]
+pub fn recursive_ownership_change(
+    path: &Path,
+    uid: Option<Uid>,
```

Comment:
It seems the `uid` is not used, is it only a placeholder?

Reply:
Was hoping to make `recursive_ownership_change` generic that it can also be used for changing `uid` so basically the input parameter to this function matches with [nix](https://docs.rs/nix/0.9.0/nix/index.html)::[unistd](https://docs.rs/nix/0.9.0/nix/unistd/index.html)::[chown](https://docs.rs/nix/0.9.0/nix/unistd/fn.chown.html) input parameter.